openwisp_controller/config/api/urls.py

```diff
@@ -0,0 +1,55 @@
+from django.conf import settings
+from django.urls import path
+
+from . import views as api_views
+
+app_name = 'openwisp_controller'
+
+
+def get_api_urls(api_views):
+    """
+    returns:: all the API urls of the config app
+    """
+    if getattr(settings, 'OPENWISP_CONTROLLER_API', True):
+        return [
+            path(
+                'controller/template/',
+                api_views.template_list,
+                name='api_template_list',
+            ),
+            path(
+                'controller/template/<str:pk>/',
+                api_views.template_detail,
+                name='api_template_detail',
+            ),
+            path(
+                'controller/template/<str:pk>/configuration/',
+                api_views.download_template_config,
+                name='api_download_template_config',
+            ),
+            path('controller/vpn/', api_views.vpn_list, name='api_vpn_list',),
```

Comment:
|    | controller/vpn/                                                                  | Status | Fail Reason                                                                                                                                                   |
|----|----------------------------------------------------------------------------------|--------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|
| 1  | [SU] Can view any VPN                                                            | Pass   |                                                                                                                                                               |
| 2  | [O1A] Can view only org1 VPN (not even shared org)                               | Pass   |                                                                                                                                                               |
| 3  | [O1O] Cannot view any VPN                                                        | Pass   |                                                                                                                                                               |
| 4  | [SU][O1A] Field validation error on field                                        | Fail   | { "config" : ["This field cannot be blank."], "__all__" :["Invalid configuration triggered by \"#/\", validator says:\n\n'openvpn' is a required property"] } |
| 5  | [SU] Only valid config schema accepted                                           | Pass   |                                                                                                                                                               |
| 6  | [SU][O1A] Can create VPN                                                         | Pass   |                                                                                                                                                               |
| 7  | [SU][O1A] Cannot create VPN                                                      | Pass   |                                                                                                                                                               |
| 8  | [SU][O1A] Unique name per org check                                              | Pass   |                                                                                                                                                               |
| 9  | [SU][O1A] Appropriate CA/Certs visible                                           | Fail   | O1A does not see shared CA/Certs                                                                                                                              |
| 10 | [01A] Trying to create device with another org's id gives object not found error | Pass   |                                                                                                                                                               |

Reply:
I'm afraid "the invalid configuration triggered by ... " is defined at model level and we have to leave as is for now.